src/bpt/temp.nix.cpp

```diff
@@ -1,12 +1,28 @@
 #include "./temp.hpp"
 
 #ifndef _WIN32
+
+#include <vector>
+
 using namespace bpt;
 
+namespace {
+
+auto string_to_vector(std::string const& s)
+{
+    std::vector<char> v(s.size() + 1);
+    std::copy_n(s.c_str(), s.size() + 1, v.begin());
+    return v;
+}
+
+} // namespace
+
 temporary_dir temporary_dir::create_in(path_ref base) {
     auto file = (base / "bpt-tmp-XXXXXX").string();
 
-    const char* tempdir_path = ::mktemp(file.data());
+    // mkdtemp() modifies the input buffer.
+    auto templ = string_to_vector(file);
+    const char* tempdir_path = ::mkdtemp(templ.data());
```

Comment:
As long as the `std::string` is non-`const`, you can call `.data()` to get a pointer-to-modifiable `char`, which should work here without needing to create a vector of characters.

This also seems to fail if the parent directory does not already exist. Add a call to `fs::create_directories(base)` beforehand and that should mitigate the issue.

It also seems there is a missing `#include <unistd.h>` for macOS to see `mkdtemp()`.